test/mock-pubsub-test-only-features.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { PubSub } from '../src/mock-pubsub';
+
+describe('test only features', () => {
+  it('enable message inspection', async () => {
+    const inspectMessage = jest.fn((m) => m);
+    const pubsub = new PubSub(
+      { projectId: 'project-id-1' },
+      {
+        interceptMessage: inspectMessage,
```

Comment:
Typing approach: documentation currently suggest to cast the mocked pub sub class with `@google-cloud/pubsub` types which don't support the new testing object argument. How could we support it?